### PR TITLE
Disable public visibility

### PR DESCRIPTION
**Pull request by thaker-nish**
https://trello.com/c/EIs1WaWL

- Added if statement to show method for companies controller.
- Added before_action authenticate_company! to companies controller.
- Added if statement to show method for customers controller.
- Added before_action authenticate_customer! to customer controller.
- Added Profile link to my Account Drop Down in application.html.erb for companies.

### CODE REVIEW

**Comment by gatorjuice** [comment]
@thaker-nish this refactor broke a bunch of tests. So you'll need to rewrite what we expect to happen in the tests.

**Comment by thaker-nish** [comment]
@gatorjuice ok I'll take a look at it

File: spec/controllers/companies_controller_spec.rb [hunk 1]
```diff
@@ -41,14 +41,25 @@
...
       expect(response).to render_template("show.html.erb")
     end
+    it 'redirects a user trying to access another users show page' do
+      company1 = create(:company,
+      email: "test@gmail.com", id: 20)
+      company = create(:company,
+      email: "example@gmail.com", id: 100)
+      sign_in company
+      get :show, params: { id: 20 }
+      expect(:company).to redirect_to("/")
```
Comment (gatorjuice) [inline spec/controllers/companies_controller_spec.rb]: no tsure why this is passing but it should be: `expect(response).to redirect_to("/")`

File: spec/controllers/customers_controller_spec.rb [hunk 2]
```diff
@@ -3,9 +3,20 @@
 RSpec.describe CustomersController, type: :controller do
   describe 'GET #show' do
     it 'assigns the requested customer to @customer' do
-      customer = create(:customer)
+      customer = create(:customer,
```
Comment (gatorjuice) [inline spec/controllers/customers_controller_spec.rb]: assigning the email address here is extraneous. It should be removed

File: spec/controllers/customers_controller_spec.rb [hunk 3]
```diff
@@ -3,9 +3,20 @@
...
       expect(assigns(:customer)).to eq(customer)
     end
+    it 'redirects a user trying to access another users show page' do
+      customer1 = create(:customer,
+      email: "test@gmail.com", id: 20)
+      customer = create(:customer,
+      email: "example@gmail.com", id: 100)
+      sign_in customer
+      get :show, params: { id: 20 }
+      expect(:customer).to redirect_to("/")
```
Comment (gatorjuice) [inline spec/controllers/customers_controller_spec.rb]: `expect(response)...`

**Comment by thaker-nish** [comment]
@gatorjuice ok, changes are up now

**Comment by gatorjuice** [comment]
👍

**Pull request merged**